src/toil/job.py

```diff
@@ -601,7 +601,7 @@ def asyncUpdate():
                     
                     #Indicate any files that should be deleted once the update of 
                     #the job wrapper is completed.
-                    self.jobWrapper.filesToDelete = len(self.filesToDelete)
+                    self.jobWrapper.filesToDelete = self.filesToDelete
```

Comment:
This should be self.jobWrapper.filesToDelete = list(self.filesToDelete), as, intentionally, the fileStore uses a set, whereas the jobWrapper uses a list. I'm fixing this in my PR on the fileStore, suggest dropping this change.